map.go

```diff
@@ -22,12 +22,12 @@ import "sync"
 // NOTE(x): In case of operating on big amounts of data or need of extended functionality - consider to use https://github.com/streamrail/concurrent-map
 type ConcurrentMap struct {
 	items map[interface{}]interface{}
-	lock  *sync.RWMutex
+	lock  sync.RWMutex
```

Comment:
To let `new(ConcurrentMap)` initialize 'ready to go' structure. In case it is still a pointer, user will get 'nil pointer dereference' panic operating on concurrent map initialized like new(ConcurrentMap).
One more argument: ConcurrentMap is publick member, and user is allowed to initialize it like new(ConcurrentMap). From my perspective there is no point to prevent it, and force user to use only factory. 
So, both ways - lets make users life simplier)))